Suggest some small tweaks to the repo

josh-arnold-1 opened the pull request:
Suggest some small tweaks, each change separated by commit.

josh-arnold-1 committed:
Update .gitignore to exclude '.vscode/' folder

josh-arnold-1 committed:
Update .gitignore to exclude '.vscode/' folder

josh-arnold-1 committed:
Add '.sourcekit-lsp/config.json' to prioritize swift packet manager for contributors using Cursor/VSCode

josh-arnold-1 committed:
Rename apple.json to config.json for more semantic clarity

josh-arnold-1 committed:
Add sourcekit lsp config.json to the Example project

josh-arnold-1 committed:
Add 'Stream SourceKit Logs' to example app

Add this so we can easily stream sourcekit logs for the example app from within VSCode

josh-arnold-1 committed:
Add symlink_prefix build flag to Example app

Add this flag to show that the BSP still works, even with a custom symlink prefix flag.

rockbruno commented on .gitignore:
```diff
@@ -9,4 +9,5 @@ Package.resolved
-__pycache__
\ No newline at end of file
+__pycache__
+.vscode/
```
```suggestion
/.vscode/
```

We can ignore the root one (auto generated by sourcekit-lsp), but the example one is currently built manually so I think we should keep that one for now

With this you should also be able to delete the root .vscode folder with this PR :)

rockbruno commented on .sourcekit-lsp/config.json:
```diff
@@ -0,0 +1,5 @@
+  "backgroundIndexing": true,
+  "backgroundPreparationMode": "build",
+  "defaultWorkspaceType": "swiftPM"
+}
```
Nice, didn't know about this one.

rockbruno commented on Example/.bazelrc:
```diff
@@ -18,6 +18,7 @@ common --features=swift.use_global_index_store
 common --features=swift.use_global_module_cache
 build --noworker_sandboxing
 build --spawn_strategy=remote,worker,local
+build --symlink_prefix=custom-out/
```
Think this might actually not do anything in practice because the BSP-related builds are configured to not generate these further below: `--experimental_convenience_symlinks=ignore`

josh-arnold-1 replied:
Thanks, I will drop this commit for now!

rockbruno reviewed (commented):
Nice stuff, thanks @josh-arnold-1 ! All looks good, just some small nits:

josh-arnold-1 committed:
Fix aquery parsing

I noticed, the aquery parsing can run into an issue where a path contains both "external/" and "bazel-out/" — this commit refactors this logic to be more readable and use `continue` statements. We also check bazel-out/ first, as that would always precede external if both are used.
